File: Python/分解质因数.py
import sys
import math
def ff(x):
	a=math.sqrt(x)
	k=2
	if x<2 :
		return -1
	else :
		while k<a:
			if x%k==0 :
				return 0
			k+=1
		return 1
x=int(sys.argv[1])
y=x//2
with open('result.txt','w') as fp:
	if x<2 :
		fp.write('I')
	elif ff(x) :
		fp.write(str(x))
	else :
		for i in range(2,y):
			while x%i==0 :
				fp.write(str(i)+' ')
				x//=i

File: C/分解质因数.c
#include<stdio.h>
#include<stdlib.h>
#include<math.h>
int ff(int x)
{
	int a,k;
	a=sqrt(x);
	if(x<2) return -1;
	for(k=2;k<=a;k++)
		if(x%k==0) return 0;
	return 1;
}
int main(int argc,char** argv)
{
	FILE *fp;
	if((fp=fopen("result.txt","w"))==NULL)
		return -1;
	int n,i,m;
	n=atoi(argv[1]);
	m=n/2;
	if(n<2) fprintf(fp,"I");
	else if(ff(n)) fprintf(fp,"%d",n);
	else
	{
		for(i=2;i<m;i++)
		{
			while(n%i==0)
			{
				fprintf(fp,"%d ",i);
				n/=i;
			}
		}
	}
	fclose(fp);	
	return 0;
}
